module/VuFind/src/VuFind/Controller/OverdriveController.php

```diff
@@ -0,0 +1,211 @@
+        
+        $view = $this->createViewModel(
+            [
+              'layout' => 'simple',
+              'ids' => $ids,
+              'action' => $action,
+              'result' => $result,
+            ]
+        );
+
+        $view->setTemplate('RecordDriver/SolrOverdrive/status-full');
+        return $view;  
+    }
+
+    /**
+     * 
+     * holdAction
+     * Hold Action handles all of the actions coming from the lightbox on the
+     * overdrive content page.  Including checkout, hold, cancel hold etc.
+     *
+     * @return array|bool|\Zend\View\Model\ViewModel
+     */
+    public function holdAction()
+    {
+        $this->debug("ODC Hold action");
+        
+        if (!is_array($patron = $this->catalogLogin())) {
+            return $patron;
+        }
+        $this->debug("patron: ".print_r($patron, true));
+        //TODO Check patron eligibility
+        //$driver->checkPatronAccess();
+        
+        $od_id = $this->params()->fromQuery('od_id');
+        $rec_id = $this->params()->fromQuery('rec_id');
+        $action = $this->params()->fromQuery('action');
+        
+        //place hold action comes in through the form
+        if (null !== $this->params()->fromPost('placeHold')) {
+            $action = "placeHold";
+        }
+        
+        $this->debug("ODRC od_id=$od_id rec_id=$rec_id action=$action");
+        //load the Record Driver.  Should be a SolrOverdrive  driver.
+        $driver = $this->serviceLocator->get('VuFind\Record\Loader')->load($rec_id);
+                    $formats = $driver->getDigitalFormats();
+            $title = $driver->getTitle();
+            $cover = $driver->getThumbnail('medium');
+        if (!$action) {
+            //double check the availability in case it 
+            //has changed since the page was loaded.
+            $avail = $driver->getOverdriveAvailability();
+            //if($avail->copiesOwned >0){
+            if ($avail->copiesAvailable>0) {
+                $action = "checkout";
+            } else {
+                $action = "hold";
+            }
+            //}else{
+                //create some kind of notification to user that something went wrong
+            //}
+
+        } elseif ($action=="placeHold") {
+            $email = $this->params()->fromPost('email');
+            $this->debug("placing Hold through OD now using email: $email");
+            $result = $this->connector->placeOverDriveHold($od_id, $patron, $email);
+        } elseif ($action=="cancelHold") {
+            $this->debug("canceling Hold through OD now");
+            $result = $this->connector->cancelHold($od_id, $patron);
+        } elseif ($action=="doCheckout") {
+            $this->debug("doing Checkout through OD now");
+            $result = $this->connector->doOverdriveCheckout($od_id, $patron);
+        } else {
+            $this->debug("action not defined: $action");
+        }
+        
+        $view = $this->createViewModel(
+            [
```

Comment:
Just a handy shortcut to know about -- when creating arrays like this one, you can use the compact function. In this case, `compact('od_id', 'rec_id', 'action', 'result', 'formats', 'cover', 'title')` will do the job. (Not that you have to do it that way -- but I like shaving off lines of code when I can get away with it.